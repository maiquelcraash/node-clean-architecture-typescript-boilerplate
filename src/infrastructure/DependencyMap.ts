import { Config } from './adapters/Config.js'
import { Router } from './adapters/http-server/Router.js';
import { IRouter } from '../interface-adapters/ports/http-server/IRouter.js';
import { IAdapter } from '../interface-adapters/ports/IAdapter.js';
import { IConfig } from '../interface-adapters/ports/IConfig.js';
import { IPort } from '../interface-adapters/ports/IPort.js';

export const dependencyMap: Map<IPort, IAdapter> = new Map([
    [IConfig, Config],
    [IRouter as IPort, Router as IAdapter],
]);
