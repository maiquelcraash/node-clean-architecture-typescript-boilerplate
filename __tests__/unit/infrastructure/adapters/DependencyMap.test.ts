import { dependencyMap } from '../../../../src/infrastructure/DependencyMap.js';

describe('infrastructure > DependencyMap', () => {
    afterEach(() => {
        jest.clearAllMocks();
        jest.restoreAllMocks();
    });

    describe('When importing dependency map', () => {
        test('then should match snapshot', () => {
            expect(Object.fromEntries(dependencyMap)).toMatchSnapshot();
        });
    });
});
