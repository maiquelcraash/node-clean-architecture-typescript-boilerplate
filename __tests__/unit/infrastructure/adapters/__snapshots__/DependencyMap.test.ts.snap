// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`infrastructure > DependencyMap When importing dependency map then should match snapshot 1`] = `
{
  "class IConfig extends IPort_js_1.IPort {
    SERVER_HOST;
    SERVER_PORT;
    SERVER_SSL;
    BODY_SIZE_LIMIT;
    SERVER_SSL_CERT_PATH;
    SERVER_SSL_PRIVATE_KEY_PATH;
}": [Function],
  "class IRouter extends IPort_js_1.IPort {
}": [Function],
}
`;
